### PR TITLE
Remove deprecated beecrypt and NSS crypto backends

### PR DESCRIPTION
**Pull request by pmatilai**
Shedding some weight to celebrate the beginning of a new cycle :fireworks:

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
:rocket: :+1:

**Comment by pmatilai** [comment]
Flushing BDB down the same drain is really, really, really tempting :innocent: but maybe not *just* yet...

**Comment by ignatenkobrain** [comment]
@pmatilai I think we need to deprecate it in 4.17 and ditch it in 4.18 while keeping bdb_ro only.

**Comment by pmatilai** [comment]
Actually BDB too is already marked deprecated in 4.16 (commit fc0169eb03c893d63dc44f2ada954d42e0e759ed)

**Comment by ignatenkobrain** [comment]
@pmatilai oh, in that case - I would ditch bdb backend and possibly enable bdb_ro by default for 4.17 and then in 4.18 disable it by default.

File: INSTALL [hunk 1]
```diff
@@ -14,24 +14,13 @@ The source for the file utility + library is available from
     ftp://ftp.astron.com/pub/file/
 
 You will need a cryptographic library to support digests and signatures.
-This library may be libgcrypt, Mozilla NSS, OpenSSL or beecrypt.
+This library may be libgcrypt, Mozilla NSS or OpenSSL.
 Which library to use can be specified with the
---with-crypto=[libgcrypt|beecrypt|nss|openssl] argument to configure,
+--with-crypto=[libgcrypt|nss|openssl] argument to configure,
```
Comment (voxik) [inline INSTALL:19]: Shouldn't be the NSS references removed similarly to beecrypt?

Comment (pmatilai) [reply INSTALL:19]: Oh, of course. Thanks for spotting!

**Comment by Conan-Kudo** [comment]
@pmatilai Come on, drop BDB! Go for the gold! 🥇

File: INSTALL [hunk 2]
```diff
@@ -14,24 +14,13 @@ The source for the file utility + library is available from
     ftp://ftp.astron.com/pub/file/
 
 You will need a cryptographic library to support digests and signatures.
-This library may be libgcrypt, Mozilla NSS, OpenSSL or beecrypt.
+This library may be libgcrypt, Mozilla NSS or OpenSSL.
```
Comment (Conan-Kudo) [inline INSTALL:17]: This should be `libgcrypt or OpenSSL`, ne?

Comment (pmatilai) [reply INSTALL:17]: Of course. Thanks for spotting.

**Comment by pmatilai** [comment]
There's at least one thing that needs to be dealt with one way or the other before dropping BDB can be seriously considered:
```
[pmatilai🎩︎lumikko rpm]$ grep %_db_backend macros.in 
%_db_backend	      bdb
```

**Comment by pmatilai** [comment]
Anyway, the DB discussion is a separate topic. Thanks for the doc review guys, fixed in the last push.

**Pull request merged**